firmware/third_party/CMakeLists.txt

```diff
@@ -42,6 +42,18 @@ file(GLOB_RECURSE SBG_ECOM_SRCS
 )
 list(APPEND SHARED_ARM_BINARY_X86_INCOMPATIBLE_THIRD_PARTY_SRCS ${SBG_ECOM_SRCS})
 
+# auto pull the 2.8 release of littlefs 
+CPMAddPackage(
+  NAME littlefs
+  GITHUB_REPOSITORY littlefs-project/littlefs
+  GIT_TAG v2.8.0
+  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/littlefs  # Specify the source directory
+)
+
+list(APPEND SHARED_ARM_BINARY_X86_INCOMPATIBLE_THIRD_PARTY_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/littlefs)
+file(GLOB_RECURSE LITTLEFS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/littlefs/lfs.c" "${CMAKE_CURRENT_SOURCE_DIR}/littlefs/lfs_util.c" )
+list(APPEND SHARED_ARM_BINARY_X86_INCOMPATIBLE_THIRD_PARTY_SRCS ${LITTLEFS_SRCS})
+
```

Comment:
You might want to do `littlefs` in another PR since this PR is already massive

Reply:
I made another PR for `littlefs` and merged it to master. This one is rebased.